test/runtests.jl

```diff
@@ -572,6 +572,17 @@ if pkgversion(ArrowTypes) >= v"2.0.2"
     @test isequal(a.x, t.x)
 end
 
+# https://github.com/apache/arrow-julia/issues/414
+df = DataFrame(("$i" => rand(1000) for i in 1:65536)...)
+Arrow.write("df.arrow", df)
+df_load = Arrow.Table("df.arrow")
+@test Tables.schema(df) == Tables.schema(df_load)
+for (col1, col2) in zip(Tables.columns(df), Tables.columns(df_load))
+    @test col1 == col2
+end
+rm("df.arrow")
```

Comment:
We can roundtrip through serialization without actually writing to disk by using the helper `Arrow.tobuffer`:
```suggestion
df_load = Arrow.Table(Arrow.tobuffer(df))
@test Tables.schema(df) == Tables.schema(df_load)
for (col1, col2) in zip(Tables.columns(df), Tables.columns(df_load))
    @test col1 == col2
end
```